SamplePay/app/src/main/java/com/example/android/samplepay/model/PaymentParams.kt

```diff
@@ -64,6 +65,7 @@ data class PaymentParams(
     /**
      * The total amount of the checkout.
      */
+    // TODO Why is this optional
```

Comment:
Good question. I don't think this should be optional here.

The only time that `PaymentRequest` does not require a total amount is in the case of selling digital goods inside of app stores, where the price is determined by the server based a SKU. That should not apply here, however. So, I believe that the total amount should be non-optional.

Reply:
We can replicate the same nullability constrains on the other end. Is the field [non–]nullable on the other end?

Reply:
It sounds like keeping it nullable is appropriate for now (at least as long as the non/zero-amount use case still applies).